presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupQueryLimits.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.resourceGroups;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public final class ResourceGroupQueryLimits
+{
+    private final Optional<Duration> executionTimeLimit;
+    private final Optional<DataSize> totalMemoryLimit;
+    private final Optional<Duration> cpuTimeLimit;
+
+    private static final ResourceGroupQueryLimits NO_LIMITS = new ResourceGroupQueryLimits(Optional.empty(), Optional.empty(), Optional.empty());
+
+    @JsonCreator
+    public ResourceGroupQueryLimits(
+            @JsonProperty("executionTimeLimit") Optional<Duration> executionTimeLimit,
+            @JsonProperty("totalMemoryLimit") Optional<DataSize> totalMemoryLimit,
+            @JsonProperty("cpuTimeLimit") Optional<Duration> cpuTimeLimit)
+    {
+        this.executionTimeLimit = requireNonNull(executionTimeLimit, "executionTimeLimit is null");
+        this.totalMemoryLimit = requireNonNull(totalMemoryLimit, "totalMemoryLimit is null");
+        this.cpuTimeLimit = requireNonNull(cpuTimeLimit, "perQueryCpuTime limit is null");
+    }
+
+    public static ResourceGroupQueryLimits createNoLimits()
+    {
+        return NO_LIMITS;
+    }
```

Comment:
As I noted above - we don't need the function if we just make `NO_LIMITS` public